src/plot.js

```diff
@@ -96,13 +125,69 @@ export function plot(options = {}) {
-    const node = mark.render(index, scales, values, dimensions, axes);
-    if (node != null) svg.appendChild(node);
+  // When faceting, render axes for fx and fy instead of x and y.
+  const {fx, fy} = scales;
+  const axisY = axes[facets !== undefined && fy ? "fy" : "y"];
+  const axisX = axes[facets !== undefined && fx ? "fx" : "x"];
+  if (axisY) svg.appendChild(axisY.render(null, scales, dimensions));
+  if (axisX) svg.appendChild(axisX.render(null, scales, dimensions));
+
+  // Render (possibly faceted) marks.
+  if (facets !== undefined) {
+    const fyDomain = fy && fy.domain();
+    const fxDomain = fx && fx.domain();
+    const fyMargins = fy && {marginTop: 0, marginBottom: 0, height: fy.bandwidth()};
+    const fxMargins = fx && {marginRight: 0, marginLeft: 0, width: fx.bandwidth()};
+    const subdimensions = {...dimensions, ...fxMargins, ...fyMargins};
+    const indexByFacet = facetMap(facetChannels);
+    facets.forEach(([key], i) => indexByFacet.set(key, i));
+    select(svg)
+        .call(g => {
+          if (fy && axes.y) {
+            const axis1 = axes.y, axis2 = nolabel(axis1);
+            const j = axis1.labelAnchor === "bottom" ? fyDomain.length - 1 : axis1.labelAnchor === "center" ? fyDomain.length >> 1 : 0;
+            g.selectAll()
+              .data(fyDomain)
+              .enter()
+              .append((ky, i) => (i === j ? axis1 : axis2).render(
+                fx && where(fxDomain, kx => indexByFacet.has([kx, ky])),
+                scales,
+                {...dimensions, ...fyMargins, offsetTop: fy(ky)}
+              ));
+          }
+          if (fx && axes.x) {
+            const axis1 = axes.x, axis2 = nolabel(axis1);
+            const j = axis1.labelAnchor === "right" ? fxDomain.length - 1 : axis1.labelAnchor === "center" ? fxDomain.length >> 1 : 0;
+            const {marginLeft, marginRight} = dimensions;
+            g.selectAll()
+              .data(fxDomain)
+              .enter()
+              .append((kx, i) => (i === j ? axis1 : axis2).render(
+                fy && where(fyDomain, ky => indexByFacet.has([kx, ky])),
+                scales,
+                {...dimensions, ...fxMargins, labelMarginLeft: marginLeft, labelMarginRight: marginRight, offsetLeft: fx(kx)}
+              ));
+          }
+        })
+        .call(g => g.selectAll()
+          .data(facetKeys(scales).filter(indexByFacet.has, indexByFacet))
+          .enter()
+          .append("g")
+            .attr("transform", facetTranslate(fx, fy))
+            .each(function(key) {
+              const j = indexByFacet.get(key);
+              for (const [mark, {channels, values, index, faceted}] of stateByMark) {
+                const renderIndex = mark.filter(faceted ? index[j] : index, channels, values);
+                const node = mark.render(renderIndex, scales, values, subdimensions);
+                if (node != null) this.appendChild(node);
+              }
+            }));
```

Comment:
This pattern makes the code a bit hard to read with the two successive anonymous functions. Suggestion:
```suggestion
    if (fy && axes.y) {
      const axis1 = axes.y, axis2 = nolabel(axis1);
      const j = axis1.labelAnchor === "bottom" ? fyDomain.length - 1 : axis1.labelAnchor === "center" ? fyDomain.length >> 1 : 0;
      select(svg).selectAll()
        .data(fyDomain)
        .enter()
        .append((ky, i) => (i === j ? axis1 : axis2).render(
          fx && where(fxDomain, kx => indexByFacet.has([kx, ky])),
          scales,
          {...dimensions, ...fyMargins, offsetTop: fy(ky)}
        ));
    }
    if (fx && axes.x) {
      const axis1 = axes.x, axis2 = nolabel(axis1);
      const j = axis1.labelAnchor === "right" ? fxDomain.length - 1 : axis1.labelAnchor === "center" ? fxDomain.length >> 1 : 0;
      const {marginLeft, marginRight} = dimensions;
      select(svg).selectAll()
        .data(fxDomain)
        .enter()
        .append((kx, i) => (i === j ? axis1 : axis2).render(
          fy && where(fyDomain, ky => indexByFacet.has([kx, ky])),
          scales,
          {...dimensions, ...fxMargins, labelMarginLeft: marginLeft, labelMarginRight: marginRight, offsetLeft: fx(kx)}
        ));
    }
    select(svg).selectAll()
      .data(facetKeys(scales).filter(indexByFacet.has, indexByFacet))
      .enter()
      .append("g")
        .attr("transform", facetTranslate(fx, fy))
        .each(function(key) {
          const j = indexByFacet.get(key);
          for (const [mark, {channels, values, index, faceted}] of stateByMark) {
            const renderIndex = mark.filter(faceted ? index[j] : index, channels, values);
            const node = mark.render(renderIndex, scales, values, subdimensions);
            if (node != null) this.appendChild(node);
          }
        });
```

Reply:
True, the _selection_.call is unnecessary. We can use a local variable instead if we don’t want to use chaining.